src/compareVersions.js

```diff
@@ -0,0 +1,22 @@
+/* exported jQueryVersionSince, compareVersions */
+
+// Returns 0 if v1 == v2, -1 if v1 < v2, 1 if v1 > v2
+function compareVersions( v1, v2 ) {
+	var rVersionParts = /^(\d+)\.(\d+)\.(\d+)/,
+		v1p = rVersionParts.exec( v1 ) || [ ],
+		v2p = rVersionParts.exec( v2 ) || [ ];
+
+	for ( var i = 1; i <= 3; i++ ) {
+		if ( +v1p[ i ] > +v2p[ i ] ) {
```

Comment:
This isn't a hot code path, but given the choice to add explicit number casting here, I figured I'd run some benches anyway. I was expecting to conclude how much faster it is. Especially in Chrome given the history of optimising for asmjs and more generally with the high "type" confidence.

To my surprise, it turned the other way around. Implicit number casting consistently performs _faster_ than explicit number casting. Up to 40X even.

I do apologize for the below table's size. It's was more an exercise in curiosity of comparing the absolute performance of the browsers and devices involved. In terms of the relative comparison I started with, it's pretty much the same in all browsers (small difference in Firefox, medium difference in Edge, huge leap in Safari).

<https://esbench.com/bench/5ab88567f2949800a0f61964>

| | Explicit | Implicit | Change
|--|--|--|--
| Chrome 65 | 14M ops/s | 110M ops/s | x8
| Chrome 67 canary | 14M ops/s | 109M ops/s | x7
| Firefox 58.0.2 | 3.9M ops/s | 4.5 ops/s  | +15%
| FirefoxNightly 60.0a1 nightly | 3.7M ops/s | 4.1M ops/s | +10%
| FirefoxDeveloper 60.0b6 aurora | 2.6M ops/s | 3.5M ops/s | +34%
| Safari 11.0.2 | 5M ops/s | 187M ops/s | x37
| Safari 11.2 (TP Release 52) | 4M ops/s | 157M ops/s | x40

Above locally on a MacBook, without throttling. Below from BrowserStack.

| | Explicit | Implicit
|--|--|--
| IE 11 (Windows 10) | 3.6M ops/s | 11.1M ops/s
| Edge 40 (EdgeHTML 15; Windows 10) | 3.7M ops/s | 7.0M ops/s
| Android 4.4 (Chrome; Galaxy S4) | 1.9M op/s | 4.1M ops/s
| Android 7.0 (Chrome; Galaxy S8) | 7M ops/s | 14M ops/s

Reply:
The Safari and Chrome numbers look so high that I think it just optimized away the empty `if` statements completely for those cases. Not sure why it wouldn't do so for the explicit ones too.

Reply:
@dmethvin Good point. However, am I correct in understanding you added the explicit casting as means to improve performance? Even without the 40X difference, or if it were 40X the other way around, it seems fast enough both ways (millions per second), to not be worth optimising with an explicit cast. Or can behaviour differ between the two in ways we want to support?

Reply:
Thanks, that makes sense :)